lib/explorer/data_frame.ex

```diff
@@ -568,13 +588,29 @@ defmodule Explorer.DataFrame do
   def select(df, columns, keep_or_drop) when is_list(columns) do
     column_names = names(df)
 
+    columns =
+      if Enum.all?(columns, &is_number/1) do
+        map_with_names =
+          Enum.zip(0..length(column_names), column_names)
+          |> Enum.into(%{})
```

Comment:
Just a minor suggestion: you could use [`Enum.with_index/2`](https://hexdocs.pm/elixir/Enum.html#with_index/2) and the [`for`](https://hexdocs.pm/elixir/Kernel.SpecialForms.html#for/1) comprehension to make this slightly shorter.

```suggestion
        map_with_names =
          for {name, idx} <- Enum.with_index(column_names), into: %{}, do: {idx, name}
```